relay-server/src/actors/envelopes.rs

```diff
@@ -1806,7 +1806,24 @@ impl UpstreamRequest for SendEnvelope {
                 Box::new(future)
             }
             Err(error) => {
-                sender.map(|sender| sender.send(Err(error)));
+                match error {
```

Comment:
Would this be the same as

```rust
if let UpstreamRequestError::RateLimited(upstream_limits) = &error {
    ProjectCache::from_registry().do_send(UpdateRateLimits::new(
        self.project_key,
        upstream_limits.clone().scope(&self.scoping),
    ));
}
if let Some(sender) = sender {
    sender.send(Err(error)).ok();
}
```

?

Reply:
Yes it would, I agree that using `map` on an `Option<T>` to execute code if the option is not empty is rather obscure and questionable style, It can be useful if you chain it in multiple operations but as it is used it only obscures what is happening.
I'll change it.